gcloud-java-bigquery/src/main/java/com/google/gcloud/bigquery/JobInfo.java

```diff
@@ -0,0 +1,305 @@
+      this.id = id;
+      return self();
+    }
+
+    /**
+     * Sets the job identity.
+     */
+    public B jobId(JobId jobId) {
+      this.jobId = jobId;
+      return self();
+    }
+
+    B selfLink(String selfLink) {
+      this.selfLink = selfLink;
+      return self();
+    }
+
+    B status(JobStatus status) {
+      this.status = status;
+      return self();
+    }
+
+    B statistics(StatisticsT statistics) {
+      this.statistics = statistics;
+      return self();
+    }
+
+    B userEmail(String userEmail) {
+      this.userEmail = userEmail;
+      return self();
+    }
+
+    /**
+     * Sets whether the job has to be dry run or not. If set, the job is not executed. A valid query
+     * will return a mostly empty response with some processing statistics, while an invalid query
+     * will return the same error it would if it wasn't a dry run. Behavior of non-query jobs is
+     * undefined.
+     */
+    public B dryRun(Boolean dryRun) {
+      this.dryRun = dryRun;
+      return self();
+    }
+
+    public abstract T build();
+  }
+
+  protected JobInfo(Builder<? extends JobInfo, ? extends StatisticsT, ?> builder) {
+    this.jobId = builder.jobId;
+    this.etag = builder.etag;
+    this.id = builder.id;
+    this.selfLink = builder.selfLink;
+    this.status = builder.status;
+    this.statistics = builder.statistics;
+    this.dryRun = builder.dryRun;
+    this.userEmail = builder.userEmail;
+  }
+
+  /**
+   * Returns the hash of the job resource.
+   */
+  public String etag() {
+    return etag;
+  }
+
+  /**
+   * Returns an opaque id for the job.
+   */
+  public String id() {
+    return id;
+  }
+
+  /**
+   * Returns the job identity.
+   */
+  public JobId jobId() {
+    return jobId;
+  }
+
+  /**
+   * Returns an URL that can be used to access the resource again. The returned URL can be used for
+   * GET requests.
+   */
+  public String selfLink() {
+    return selfLink;
+  }
+
+  /**
+   * Returns the status of this job. Examine this value when polling an asynchronous job to see if
+   * the job is complete.
+   */
+  public JobStatus status() {
+    return status;
+  }
+
+  /**
+   * Returns information about the job, including starting time and ending time of the job.
+   */
+  public StatisticsT statistics() {
+    return statistics;
+  }
+
+  /**
+   * Returns the email address of the user who ran the job.
+   */
+  public String userEmail() {
+    return userEmail;
+  }
+
+  /**
+   * Returns whether the job has to be dry run or not. If set, the job is not executed. A valid
+   * query will return a mostly empty response with some processing statistics, while an invalid
+   * query will return the same error it would if it wasn't a dry run. Behavior of non-query jobs is
+   * undefined.
+   */
+  public Boolean dryRun() {
+    return dryRun;
+  }
+
+  /**
+   * Returns a builder for the job.
+   */
+  public abstract Builder toBuilder();
+
+  protected MoreObjects.ToStringHelper toStringHelper() {
```

Comment:
did you consider my comment above? Also `MoreObjects.` is redundant (already imported).